mlir/lib/Target/CppOutput/gridwise_convolution_implicit_gemm_v4r4.cpp

```diff
@@ -69,38 +77,110 @@ class PopulateParams : public PopulateParamsBase {
                                        derived);
   }
 
+  LogicalResult
+  CalculateBlockGemmPerformanceParameters(InitParamsNonXDL *param,
+                                          ConvolutionContext &ctx,
+                                          DerivedBlockGemmParams &derived) {
+
+      derived.gemmMLevel0Cluster = 0;
+      derived.gemmNLevel0Cluster = 0;
+      derived.gemmMLevel1Cluster = 0;
+      derived.gemmNLevel1Cluster = 0;
+
+      if(param->blockSize == 64)
+      {
+          derived.gemmMLevel0Cluster = 4;
+          derived.gemmNLevel0Cluster = 4;
+          derived.gemmMLevel1Cluster = 2;
+          derived.gemmNLevel1Cluster = 2;
+      }
+      else if(param->blockSize == 128)
+      {
+          derived.gemmMLevel0Cluster = 4;
+          derived.gemmNLevel0Cluster = 4;
+          derived.gemmMLevel1Cluster = 4;
+          derived.gemmNLevel1Cluster = 2;
+      }
+      else if(param->blockSize == 256)
+      {
+          derived.gemmMLevel0Cluster = 4;
+          derived.gemmNLevel0Cluster = 4;
+          derived.gemmMLevel1Cluster = 4;
+          derived.gemmNLevel1Cluster = 4;
+      }
+      else
+      {
+          return failure();
+      }
+
+      if(!(param->gemmMPerThread >= 2 && param->gemmMPerThread <= 4))
+          return failure();
+
+      if(!(param->gemmNPerThread >= 2 && param->gemmNPerThread <= 4))
+          return failure();
+
+      if(!(param->gemmMPerBlock % param->gemmMPerThread == 0 && param->gemmNPerBlock % param->gemmNPerThread == 0))
+          return failure();
+
+      const auto thread_gemm_per_block_m = param->gemmMPerBlock / param->gemmMPerThread;
```

Comment:
MLIR style guide requires camelBack, see [here](https://mlir.llvm.org/getting_started/DeveloperGuide/#style-guide). I know we are not uniform in all cases, but let's try to stick to the convention as close as possible.

Reply:
This applies to multiple places.